cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -84,22 +82,8 @@ func init() {
 	}
 }
 
-type migInformation struct {
-	config   Mig
-	basename string
-}
-
-type machineTypeKey struct {
-	zone        string
-	machineType string
-}
-
 // GceManager handles gce communication and data caching.
 type GceManager interface {
-	// RegisterMig registers mig in GceManager. Returns true if the node group didn't exist before.
-	RegisterMig(mig Mig) bool
```

Comment:
GceManager is only used by gce_cloud_provider.go (and it only exists for testing GceCloudProvider with a mock, BTW). However, RegisterMig() was always called only by the gceManagerImpl itself. Therefore, it's perfectly useless as part of this interface, whether exported or not.

I originally intended to embed gceManagerImpl (after exporting it) in gkeManager, in order to reuse the logic as much as possible. For that, almost all of its methods and fields would have to be exported. However, given that most of the shared logic is being extracted to clients and cache, we can probably avoid embedding and create independent implementation with only a limited amount of duplicated code. Then this method could be private. WDYT?

Reply:
By that logic having anything public on GceManager is pointless (since we never use it outside of package and, hopefully, never will). My intuition would be that anything we call from different files (gce_cloud_provider.go in particular) should be public and anything else private, but apparently that's not true and it's just completely random :(

Still, we don't call it from outside manager, so if we don't need this in GKE I'm ok with it being public.